warehouse/query-core/src/main/java/datawave/query/function/LimitFields.java

```diff
@@ -180,7 +180,8 @@ private void manageHitsAndMisses(String keyWithGrouping, String keyNoGrouping, A
                         String[] commonalityAndGroupingContext = LimitFields.getCommonalityAndGroupingContext(key);
                         if (commonalityAndGroupingContext != null) {
                             String hitTermKeyCommonality = commonalityAndGroupingContext[0];
-                            if (hitTermKeyCommonality.equals(keyWithGroupingCommonality) && key.endsWith(keyWithGroupingSuffix)) {
+                            String hitTermGroup = commonalityAndGroupingContext[1];
+                            if (hitTermKeyCommonality.equals(keyWithGroupingCommonality) && keyWithGroupingSuffix.equals(hitTermGroup)) {
```

Comment:
Can we add an issue to refactor and clean up some of the extra object creation here?

Reply:
feel free to open one

Reply:
Opened https://github.com/NationalSecurityAgency/datawave/issues/1680